src/lib/ui/DeviceDetails/DeviceDetails.tsx

```diff
@@ -54,6 +61,39 @@ function LockDeviceDetails(props: { device: LockDevice; onBack?: () => void }) {
   const accessCodeLength =
     device.properties?.schlage_metadata?.access_code_length
 
+  function generateDeviceAlerts() {
+    console.log('device', device)
+    if (!device.errors && !device.warnings) {
+      return []
+    }
+
+    const alerts: AlertProps[] = []
+
+    if (device.errors) {
+      alerts.push(
+        ...device.errors.map((error) => ({
+          variant: 'error' as AlertProps['variant'],
+          message: errorCodeToMessageMapping[error.error_code] ?? error.message,
+        }))
+      )
+    }
+
+    if (device.warnings) {
+      alerts.push(
+        ...device.warnings.map((warning) => ({
+          variant: 'warning' as AlertProps['variant'],
+          message:
+            warningCodeToMessageMapping[warning.warning_code] ??
+            warning.message,
+        }))
+      )
+    }
+
+    return alerts
+  }
+
+  const alerts = generateDeviceAlerts()
+
   if (accessCodes == null) {
```

Comment:
Not my code but should this be `===` since the type of `accessCodes` is `UseAccessCodeData | undefined`? Stricter equality checks are generally better IMO